### ISSUE TITLE
Need programmatic tag functionality; bundle and invocationImage

### ISSUE THREAD

**Issue by squillace** [getporter/porter#1261]
**Is your feature request related to a problem? Please describe.**
When you build a porter.yaml file locally, you can drop the tag in with the tag: key inside the porter.yaml file. For developers, this is all well and good. But for automated iterative builds, _the tag name will never increment unless you manually change it inside the porter.yaml. This means that the precision we use to precisely identify image layers inside CNAB and porter is NOT what we do with the invocationImage itself. 

Further, doing iterative builds, the invocationImage tag will also not increment or be generated in any way, which gives the same problem for both manifest images and invocationImages: no build pipeline can automatically increment without reaching into the text file PRIOR TO building and changing the string. Meh. 

**Describe the solution you'd like**
Either the tag is optional, and the bundle is given digest identities prior; or some mechanism to inject tags and override/rewrite what's in the yaml should be considered. It's a hard problem, but someone's got to make GitOps happen.

**Describe alternatives you've considered**
no idea

**Additional context**
I could add random things, but

**Comment by MChorfa** [comment]
AS a workaround you can have a CI step where you can: 

## - generate-bundle-file: Command line to generate porter.yaml file from template.

```shell
generate-bundle-file:
@echo "====================================================="
@echo " Bundle name is: $(BUNDLE_NAME)"
@echo " Bundle version is: $(BUNDLE_VERSION)"
@echo " Registry is: $(DOCKER_REGISTRY)"
@echo " Project is: $(PROJECT_NAME)"
@echo " Branch is: $(CI_COMMIT_REF_NAME)"
@echo "====================================================="
@envsubst < "./porter.yaml.tmpl" > "porter.yaml"
```

### The template file "porter.yaml.tmpl" would look like
```yaml
name: ${BUNDLE_NAME}
version: ${BUNDLE_VERSION}${CI_COMMIT_REF_NAME}
description: "CORE Bundle that include nginx and more..."
tag: ${DOCKER_REGISTRY}/${PROJECT_NAME}-${PROJECT_COMPONENT}:${BUNDLE_VERSION}${CI_COMMIT_REF_NAME}

mixins:
  - exec
  - kubernetes
  - helm3:
      repositories:
        stable:
          url: "https://kubernetes-charts.storage.googleapis.com"

...
```

**Comment by vdice** [comment]
For full-fledged support of the mentioned CI/dynamic override use cases, see https://github.com/deislabs/porter/issues/1151

Does that proposal meet the needs mentioned here?  I think we're about ready to work on this, so please do check it out and contribute feedback.

**Comment by squillace** [comment]
@vdice here's what I do now to make this work. :-P https://github.com/squillace/gh-cnab/blob/master/.github/workflows/main.yml

**Comment by carolynvs** [comment]
This was implemented in #1151
